scripts/dockermake.sh

```diff
@@ -4,24 +4,79 @@
 # container as a specific user and group. By default, the effective
 # user and group id of the host are used.
 #
-# The environment variables SWG_DOCKER_IMAGE and SWG_DOCKER_TAG can be
+# The environment variables SWG_IMAGE and SWG_TAG can be
 # used to override the Docker image to use. For example:
 #
-# SWG_DOCKER_IMAGE=industrious-squirrel SWG_DOCKER_TAG=slim-buster scripts/dockerbuild.sh build
-
-_DEFAULT_TAG="latest"
-_DEFAULT_IMAGE="ghcr.io/scilifelabdatacentre/swg-data-builder"
+# SWG_IMAGE=industrious-squirrel SWG_TAG=slim-buster scripts/dockerbuild.sh build
 
 CWD="$(pwd)"
-mkdir -p "${DATA_DIR:=data}" "${INSTALL_DIR:=hugo/static}"
+
+: ${SWG_TAG="main"}
+: ${SWG_IMAGE:="ghcr.io/scilifelabdatacentre/swg-data-builder"}
+: ${SWG_DATA_DIR:="data"}
+: ${SWG_INSTALL_DIR:="hugo/static"}
+: ${SWG_CONFIG_DIR:="config"}
+
+_DOCKER_ARGS=()
+
+# Run make in Docker container using host Makefile, with data,
+# configuration and installation directories mounted.
+docker_make() {
+    _DOCKER_WORKDIR=/swedgene
+    # Make sure writable directories exist on the host
+    mkdir -p "$SWG_DATA_DIR" "$SWG_INSTALL_DIR"
+    docker run --rm "${_DOCKER_ARGS[@]}" \
```

Comment:
Related to the ARM64 vs AMD64 issue, I mentioned the other day,
`scripts/dockermake.sh --test` eventually runs into this error on my Macbook:

`docker: image with reference ghcr.io/scilifelabdatacentre/swg-data-builder:main was found but does not match the specified platform: wanted linux/arm64, actual: linux/amd64.`

I am able to I get around this error by adding `--platform=linux/amd64 \` here as an argument for `docker run`.
The image still gets flagged with the !AMD64 warning in my docker client, but I am now able to run `scripts/dockermake.sh --test` commands.

I'll continue testing with my modified `dockermake.sh` for now, but will not commit the change. Maybe there is a better solution than to hardcode the platform into the script?

Reply:
Until we have found a solution to perhaps build images for multiple architectures as part of our CI, I would suggest that you build your images locally. For example:
```
./scripts/dockerbuild.sh data
SWG_TAG=local ./scripts/dockermake.sh --test debug
```
The first command will build the `ghcr.io/scilifelabdatacentre/swg-data-builder:local` image on your computer. Specifying `SWG_TAG=local` will instruct `dockermake.sh` to use that one instead of the default `ghcr.io/scilifelabdatacentre/swg-data-builder:main` that is remotely built.

Reply:
Also mentioning the `SWG_DEFAULT_USER=1` environment variable you can pass to `dockerbuild.sh` if you run into user issues. This will use the default user id `1000` when building the image instead of trying to use the user ID you have on your machine. It's a bit messy still, sorry.

Reply:
Excellent, thank you so much for this advice! `SWG_DEFAULT_USER=1 scripts/dockerbuild.sh` solved the previous GID issue I had. The local image is installed (and, as expected, lacks the !AMD64 flag). I then rolled  back the `--platform` arg that I had been playing with. Using `SWG_TAG=local ./scripts/dockermake.sh --test build` works without a problem. Will continue to test other make recipes and report back!